internal/engine/eval/vulncheck/report.go

```diff
@@ -0,0 +1,331 @@
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package vulncheck provides the vulnerability check evaluator
+package vulncheck
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	htmltemplate "html/template"
+	"strings"
+	"text/template"
+)
+
+type vulnerabilityReport interface {
+	render() (string, error)
+}
+
+type minderTemplateData struct {
+	MagicComment string
+	Metadata     string
+	Body         string
+	Footer       string
+}
+
+type reportMetadata struct {
+	VulnerabilityCount int    `json:"VulnerabilityCount"`
+	RemediationCount   int    `json:"RemediationCount"`
+	TrackedDepsCount   int    `json:"TrackedDepsCount"`
+	CommitSHA          string `json:"CommitSHA"`
+}
+
+const (
+	vulnsFoundText = `
+	Minder found vulnerable dependencies in this PR. Either push an updated
+	version or accept the proposed changes. Note that accepting the changes will
+	include Minder as a co-author of this PR.
+	`
+	vulnsFoundTextShort = `
+	Vulnerable dependencies found.
+	`
+	noVulsFoundText = `
+	Minder analyzed this PR and found no vulnerable dependencies.
+	`
+	reviewBodyDismissCommentText = `
+	Previous Minder review was dismissed because the PR was updated.
+	`
+	vulnFoundWithNoPatch = "Vulnerability found, but no patched version exists yet."
+)
+
+const (
+	reviewBodyMagicComment = "<!-- minder: pr-review-body -->"
+	statusBodyMagicComment = "<!-- minder: pr-status-body -->"
+	separatorComment       = "<!--[SEP]-->"
```

Comment:
`separatorComment` seems to be unused.

Reply:
Separator is meant for adding metadata to the body of the review and allowing us to easily parse it. So we can include data from a previous review as a comment in JSON form and use it to provide a comparison to the current review. i.e the below will provide info on the previous status report:

```
<!-- minder: pr-status-body --><!--[SEP]--><!--{ {"VulnerabilityCount":1,"RemediationCount":0,"TrackedDepsCount":1,"CommitSHA":"27d6810b861c81e8c61e09c651875f5a976781d1"} }--><!--[SEP]-->

        <h2>Minder Vulnerability Report &#9888;&#65039;</h2>
...
```

Probably should have made this more clear. It was also mentioned it here https://github.com/stacklok/minder/issues/2149 :smile:

Reply:
yes, but the constant `separatorComment` is still unused, the "<!--[SEP]-->` string constant is used verbatim in  `minderTemplateString`.